lib/services/validator.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:kisgeri24/logging.dart';
+
+@Deprecated("Not needed anymore since registration won't be a feature.")
+String? validateName(String? value) {
+  String pattern = r'(^[a-zA-Z ]*$)';
+  RegExp regExp = RegExp(pattern);
+  if (value?.isEmpty ?? true) {
+    return "Name is required";
+  } else if (!regExp.hasMatch(value ?? '')) {
+    return "Name must be a-z and A-Z";
+  }
+  return null;
+}
+
+@Deprecated(
+    "This wasn't used but can be later on if we'd decide to make the phone number editable.")
+String? validateMobile(String? value) {
+  logger.i("Validating phone number: $value");
+  String pattern =
+      r'((?:\+?3|0)6)(?:-|\()?(\d{1,2})(?:-|\))?(\d{3})-?(\d{3,4})';
+  RegExp regExp = RegExp(pattern);
+  if (value == null || value.isEmpty) {
+    logger.i(
+        "The given input for phone number validation is either null or empty thus no validation can be done!");
+    return "Mobile phone number is required";
+  } else if (!regExp.hasMatch(value)) {
+    logger.i("Phone number ($value) is not acceptable.");
+    return "Mobile phone number must contain only digits";
+  }
+  logger.i("Phone number validation passed!");
+  return null;
+}
+
+String? validatePassword(String? value) {
+  if ((value?.length ?? 0) < 6) {
+    return 'Password must be more than 5 characters';
+  } else {
+    return null;
+  }
+}
+
+String? validateEmail(String? value) {
+  String pattern =
+      r'^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$';
```

Comment:
It may worth to consider to use a package instead of this RegEx. It would probably only make difference in very extreme edge cases. This package seems to be the flutter standard for email validation: [email_validator](https://pub.dev/packages/email_validator).
(Edge cases for email, if someone is interested: [https://stackoverflow.com/a/201378/8659684](https://stackoverflow.com/a/201378/8659684))

Reply:
yeahh, looks neat, gonna replace it with this package. I'm also not a huge fan of robust regexes 😅

Reply:
updated as promised